File: Ejercicio1/BicicletaCarretera.java
package Ejercicio1;

public class BicicletaCarretera extends Bicicleta{
    private static int id= 0;
    
    public BicicletaCarretera() {
        super(id);
        ++id;
    }
/* 
    @Override
    public ArrayList<Bicicleta> crearBicicletas(int N){
        ArrayList<Bicicleta> bicicletas = new ArrayList<>();
        for(int i = 0; i < N; i++){
            bicicletas.add(new BicicletaCarretera());
        }
        return bicicletas;
    }
*/   
    @Override
    public String toString() {
        return ("Bicicleta de carretera con identificador de carrera " + getIdCarrera());
    }
}

File: Ejercicio2/BicicletaCarretera.py
import Bicicleta

class BicicletaCarretera(Bicicleta):
    def __init__(self, id):
        super().init(id)
        self.id += 1

    def toString(self):
        print("Bicicleta de carretera con id: " + self.id + " ha terminado la carrera.")
              

    
